pilot/pkg/config/kube/gateway/testdata/route-precedence.yaml

```diff
@@ -0,0 +1,100 @@
+apiVersion: gateway.networking.k8s.io/v1alpha2
```

Comment:
Cases to think about is merging routes that are across namespace. Especially with `mesh`

Reply:
Yeah this is problematic..

```yaml
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: echo
  namespace: default
spec:
  parentRefs:
  - kind: Mesh
    name: istio
  hostnames: ["echo.default.svc.cluster.local"]
  rules:
  - backendRefs:
    - name: echo
      port: 80
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: echo
  namespace: consumer
spec:
  parentRefs:
  - kind: Mesh
    name: istio
  hostnames: ["echo.default.svc.cluster.local"]
  rules:
  - matches:
    - path:
        type: PathPrefix
        value: /get
    backendRefs:
    - name: echo
      port: 80
```

Ends up getting merged like:
```yaml
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parent: HTTPRoute/echo.default
  name: echo-istio-autogenerated-k8s-gateway-0
  namespace: default
spec:
  gateways:
  - mesh
  hosts:
  - echo.default.svc.cluster.local
  http:
  - match:
    - name: echo-0-PathPrefix
      uri:
        regex: /get((\/).*)?
    route:
    - destination:
        host: echo.consumer.svc.cluster.local
        port:
          number: 80
  - match:
    - name: echo-0-PathPrefix
      uri:
        prefix: /
    route:
    - destination:
        host: echo.default.svc.cluster.local
        port:
          number: 80
```

Reply:
Sorry I didn't get the problem, could you please explain the right behavior? Thanks!

Reply:
I am not quite sure - but not this I think. I'll need to think about it. I am thinking we shouldn't merge across namespace for mesh but will need to think about all cases

Reply:
Seems related to #22997 and #36621

Reply:
Ok sorry for the delay. I had a complete misunderstand of how VirtualService even works! so I think we may need some broader changes. Will investigate a bit more and get back to you

Reply:
Need a bit more time. sorry

Reply:
Apologies for the delay - can't believe this was 5 weeks! I was thinking about this the entire time believe it or not, its a lot trickier problem than I originally anticipated. I am planning to write up a full design doc on how I think Mesh routes should work, but I think for now it would be safe to make it so we just do not merge across namespaces. I think we will need more changes than that, but that is out of scope for this PR.

BTW, for Gateway case, can we add a test like:

```
Gateway:
  allowedRoutes: [ns1, ns2]
Route in ns1
Route in ns2
Route in ns3
```

Result should be ns1+ns2 merge, and ns3 is properly ignore

Reply:
No problem :)
Fixed, thanks for elaborating